src/haddock/modules/analysis/alascan/scan.py

```diff
@@ -0,0 +1,591 @@
+                    'delta_vdw': delta_vdw,
+                    'delta_elec': delta_elec,
+                    'delta_desolv': delta_desolv,
+                    'delta_bsa': delta_bsa,
+                    'frac_pr': 1
+                    }
+            else:
+                clt_scan[cl_id][ident]['delta_score'] += delta_score
+                clt_scan[cl_id][ident]['delta_vdw'] += delta_vdw
+                clt_scan[cl_id][ident]['delta_elec'] += delta_elec
+                clt_scan[cl_id][ident]['delta_desolv'] += delta_desolv
+                clt_scan[cl_id][ident]['delta_bsa'] += delta_bsa
+                clt_scan[cl_id][ident]['frac_pr'] += 1
+    # now average the data
+    for cl_id in clt_scan:
+        scan_clt_filename = f"scan_clt_{cl_id}.csv"
+        log.info(f"Writing {scan_clt_filename}")
+        clt_data = []
+        for ident in clt_scan[cl_id]:
+            chain = ident.split("-")[0]
+            resnum = int(ident.split("-")[1])
+            resname = ident.split("-")[2]
+            frac_pr = clt_scan[cl_id][ident]["frac_pr"]
+            clt_data.append([chain, resnum, resname, ident,
+                             clt_scan[cl_id][ident]['delta_score'] / frac_pr,
+                             clt_scan[cl_id][ident]['delta_vdw'] / frac_pr,
+                             clt_scan[cl_id][ident]['delta_elec'] / frac_pr,
+                             clt_scan[cl_id][ident]['delta_desolv'] / frac_pr,
+                             clt_scan[cl_id][ident]['delta_bsa'] / frac_pr,
+                             clt_scan[cl_id][ident]['frac_pr'] / cl_pops[cl_id]
+                             ]
+                            )
+        df_cols = ['chain', 'resnum', 'resname', 'full_resname', 'delta_score',
+                   'delta_vdw', 'delta_elec', 'delta_desolv', 'delta_bsa',
+                   'frac_pres']
+        df_scan_clt = pd.DataFrame(clt_data, columns=df_cols)
+        # adding clt-based Z score
+        df_scan_clt = add_zscores(df_scan_clt, 'delta_score')
+
+        df_scan_clt.sort_values(by=['chain', 'resnum'], inplace=True)
+        df_scan_clt.to_csv(
+            scan_clt_filename,
+            index=False,
+            float_format='%.2f',
+            sep="\t")
+
+        # add comment
+        fl_content = open(scan_clt_filename, 'r').read()
+        with open(scan_clt_filename, 'w') as f:
+                f.write(f"#######################################################################{os.linesep}")  # noqa E501
+                f.write(f"# `alascan` cluster results for cluster {cl_id}{os.linesep}")  # noqa E501
+                f.write(f"#{os.linesep}")
+                f.write(f"# z_score is calculated with respect to the mean values of all residues{os.linesep}")  # noqa E501
+                f.write(f"#######################################################################{os.linesep}")  # noqa E501
+                f.write(fl_content)
+    return clt_scan
+
+def check_alascan_jobs(alascan_jobs):
```

Comment:
Shouldn't this handled by the ontology lib?

Reply:
is it handled there? then this function can be removed!